airbase/download_api/concrete_api_client.py

```diff
@@ -61,59 +61,75 @@ async def __aexit__(
         exc_val: BaseException | None,
         exc_tb: TracebackType | None,
     ) -> None:
-        assert self.session is not None, "outside context manager"
-        await self.session.close()
-        self.session = self.semaphore = None
+        await self.__session.close()  # type:ignore[union-attr]
+        self.__session = self.__semaphore = None
+
+    @property
+    def _session(self) -> aiohttp.ClientSession:
+        """private `aiohttp.ClientSession`, only available inside the context manager"""
+        if self.__session is None:
+            raise RuntimeError(
+                f"{self.__class__.__qualname__}._session called outside the context manager"
+            )
+        return self.__session
+
+    @property
+    def _semaphore(self) -> asyncio.Semaphore:
+        """private `asyncio.Semaphore`, only available inside the context manager"""
+        if self.__semaphore is None:
+            raise RuntimeError(
+                f"{self.__class__.__qualname__}._semaphore called outside the context manager"
+            )
+        return self.__semaphore
 
     async def country(self) -> CountryResponse:
         """get request to /Country"""
-        assert self.session is not None, "outside context manager"
-        async with self.session.get(self.base_url + "/Country") as r:
-            r.raise_for_status()
-            return await r.json(encoding="UTF-8")  # type:ignore[no-any-return]
+        async with self._semaphore:
```

Comment:
Why do we need a semaphore here when we are already limiting the connection pool size? `aiohttp` should handle that internally.

Reply:
fair point, I'll remove the superfluous `with semaphore` calls

Reply:
done